### ISSUE TITLE
[clang] Clang does not diagnose unused private member functions

### ISSUE THREAD

**Issue by carlosgalvezp** [llvm/llvm-project#106678]
Hi!

I notice clang doesn't diagnose this unused private member function:

```cpp
class Foo
{
 private:
    void foo(){}
};

```
https://godbolt.org/z/nPTPd4dqc

I'm curious to understand why that's the case. Is there high likelihood of false positives?

Thanks!

**Comment by AaronBallman** [comment]
We diagnose unused private fields which suggests we really should handle unused private functions (and other private declarations as well). I can't think of a reason not to warn on them.
